daemon/prune.go

```diff
@@ -150,3 +154,72 @@ func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.Image
 
 	return rep, nil
 }
+
+// localNetworksPrune removes unused local networks
+func (daemon *Daemon) localNetworksPrune(config *types.NetworksPruneConfig) (*types.NetworksPruneReport, error) {
+	rep := &types.NetworksPruneReport{}
+	var err error
+	// When the function returns true, the walk will stop.
+	l := func(nw libnetwork.Network) bool {
+		nwName := nw.Name()
+		predefined := runconfig.IsPreDefinedNetwork(nwName)
+		if !predefined && len(nw.Endpoints()) == 0 {
```

Comment:
Ahh right, did not look carefully. I'd avoid checking the error string, unless there is no other option.
Then, what about logging the event here, something like:
```
	l := func(nw libnetwork.Network) bool {
		if len(nw.Endpoints()) != 0 {
			return false
		}
		err := nw.Delete()
		if err == nil {
			daemon.LogNetworkEvent(nw, "destroy")
			rep.NetworksDeleted = append(rep.NetworksDeleted, nwName)
		} else {
			logrus.Warnf("could not remove network %s: %v", nw.Name(), err)
		}
		return false
	}
```